pkg/streamingpromql/engine_fuzz_test.go

```diff
@@ -0,0 +1,270 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+// Provenance-includes-location: https://github.com/prometheus/prometheus/tree/main/promql/engine_test.go
+// Provenance-includes-license: Apache-2.0
+// Provenance-includes-copyright: The Prometheus Authors
+
+package streamingpromql
+
+import (
+	"context"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/prometheus/prometheus/promql"
+	"github.com/prometheus/prometheus/promql/promqltest"
+	"github.com/prometheus/prometheus/util/teststorage"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/grafana/mimir/pkg/querier/stats"
+	"github.com/grafana/mimir/pkg/streamingpromql/testutils"
+)
+
+// A construct to hold our query engines and storage. Also holds rules for how we process certain queries
+type TestQueryContext struct {
+	prometheusEngine  *promql.Engine
+	mimirEngine       *Engine
+	testStorage       *teststorage.TestStorage
```

Comment:
Likewise, can't we just use `storage.Queryable` interface here, because from what I can see, ultimately we just want to pass this to `NewInstantQuery` and `NewRangeQuery`.